src/realm/dictionary.cpp

```diff
@@ -612,8 +759,77 @@ void Dictionary::erase(Mixed key)
 
     if (size()) {
         auto hash = key.hash();
-        ObjKey k(int64_t(hash & 0x7FFFFFFFFFFFFFFF));
-        auto state = m_clusters->get(k);
+        ObjKey k(int64_t(hash & s_hash_mask));
+        auto state = m_clusters->try_get(k);
+        if (state.index == realm::npos) {
+            throw KeyNotFound("Dictionary::erase");
+        }
+
+        if (m_clusters->has_collisions()) {
+            Array fallback(m_obj.get_alloc());
+            Array& fields = m_clusters->get_fields_accessor(fallback, state.mem);
+            ArrayRef refs(m_obj.get_alloc());
+            refs.set_parent(&fields, 3);
+            refs.init_from_parent();
+
+            auto ref = refs.get(state.index);
+            if (ref) {
+                Array links(m_obj.get_alloc());
+                links.set_parent(&refs, state.index);
+                links.init_from_parent();
+                auto sz = links.size();
+                REALM_ASSERT(sz > 0);
+
+                if (do_get_key(state).compare_signed(key) == 0) {
+                    // We are erasing main entry, swap last sibling over
+                    ObjKey k2(links.get(sz - 1));
+                    if (links.size() == 1) {
+                        refs.set(links.get_ndx_in_parent(), 0);
+                        links.destroy();
+                    }
+                    else {
+                        links.erase(sz - 1);
+                    }
+                    auto state2 = m_clusters->get(k2);
+                    swap(state, state2);
```

Comment:
This call confused me at first glance because I thought it'd be something like std::swap() and it's not that at all.

Reply:
Renamed to `swap_content`